packages/interactor/src/index.ts

```diff
@@ -0,0 +1,121 @@
+import { converge } from './converge';
+
+interface Options {
+  timeout: number,
+  document?: HTMLDocument
+}
+
+let defaultOptions: Options = {
+  timeout: 1900
+}
+
+type ActionSpecification = Record<string, (element: HTMLElement) => unknown>
+
+type ActionImplementation<T extends ActionSpecification> = {
+  [P in keyof T]: T[P] extends ((element: HTMLElement, ...args: infer TArgs) => infer TReturn) ? ((...args: TArgs) => Promise<TReturn>) : never;
+}
+
+export interface InteractorSpecification {
+  name: string,
+  selector: string,
+  defaultLocator: (element: HTMLElement) => string;
+}
+
+export type LocatorSpecification = string;
+
+export class Interactor {
+  protected parent?: Interactor;
+
+  constructor(private specification: InteractorSpecification, private locator: LocatorSpecification) {
+  }
+
+  find<T extends Interactor>(interactor: T): T {
+    let child = Object.create(interactor);
+    child.parent = this;
+    return child;
```

Comment:
Not sure if this is acceptable or if it's too clever? Manually messing with prototypal inheritance doesn't seem to common in JS.

Reply:
Good suggestion!